wpilibc/src/main/native/include/frc/simulation/SimLinearSystem.h

```diff
@@ -0,0 +1,87 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <array>
+
+#include <Eigen/Core>
+#include <units/current.h>
+#include <units/time.h>
+
+#include "frc/StateSpaceUtil.h"
+#include "frc/system/LinearSystem.h"
+
+namespace frc {
+
+namespace sim {
+
+template <int States, int Inputs, int Outputs>
+class SimLinearSystem {
+ public:
+  /**
+   * Create a SimLinearSystem. This simulator uses an {@link LinearSystem} to
+   * simulate the state of the system. In simulationPeriodic, users should first
+   * set inputs from motors, update the simulation and write simulated outputs
+   * to sensors.
+   *
+   * @param system             The system being controlled.
+   * @param addNoise           If we should add noise to the measurement vector
+   * @param measurementStdDevs Standard deviations of measurements. Can be null
+   * if addNoise is false.
+   */
+  SimLinearSystem(frc::LinearSystem<States, Inputs, Outputs>& plant,
```

Comment:
Instead of passing a bool for whether to enable noise, have two ctors where the one for which an array of measurement standard deviations is supplied has noise automatically enabled.